modules/installation-azure-user-defined-routing.adoc

```diff
@@ -0,0 +1,72 @@
+// Module included in the following assemblies:
+//
+// * installing/installing_azure/installing-azure-private.adoc
+
+[id="installation-azure-user-defined-routing_{context}"]
+= User-defined outbound routing
+
+In {product-title}, you can choose your own outbound routing for a cluster to
+connect to the internet. This allows you to skip the creation of public IP
+addresses and public load balancers. This is required if you want to prevent
+exposing external endpoints for a private cluster.
+
+You can configure user-defined routing by modifying parameters in the
+`install-config.yaml` file before installing your cluster. A pre-existing VNet
+is required to use outbound routing when installing a cluster; the installation
+program is not responsible for configuring this.
+
+When configuring a cluster to use user-defined routing, the installation program
+does not create the following resources:
+
+* Outbound rules
+* Kubernetes Loadbalancer
+* Public load balancer configurations for MachineSets
+
+You must ensure the following items are available before setting user-defined
+routing:
+
+* Egress to the internet is possible to pull container images, unless using an
+internal registry mirror.
+* The cluster is able to access Azure APIs.
+* Various allowlist endpoints are configured. You can reference these endpoints
+in the *Configuring your firewall* section.
+
+There are several pre-existing networking setups that are supported for internet
+access using user-defined routing.
+
+[discrete]
+== Private cluster with a proxy configuration
```

Comment:
I would re-order the types of UDR Private clusters by complexity and probably use

1) NAT (simplified version of the section that is lower in the document)
2) Firewall
3) Proxy
4) No Internet Access

Possible addition:
5) Custom (requires a route for 0.0.0.0/0 and next hop destination of your router in the route table)

Reply:
I will delete the separate module for NAT and include it with the other use cases (ordered as suggested in your comment). Thanks!